rfcs/2023-01-06-layout.md

```diff
@@ -0,0 +1,461 @@
+| Flexbox             | `alignSelf`, `justifySelf`, `order`                                            | Only restricted set of properties applicable on flex children. Note: `justifySelf` doesn't work for flex layout but it can be used for block layout (web only).     |
+| Position            | `position`, `zIndex`, `top`, etc.                                              | [List of props](https://github.com/styled-system/styled-system/blob/master/docs/table.md#position) referenced from styled-system.                                   |
+| Grid **(web only)** | `gridColumn*`, `gridRow*`, `gridArea`, `justifySelf`, `alignSelf`, `placeSelf` | Only restricted set of properties applicable on [grid children](https://css-tricks.com/snippets/css/complete-guide-grid/#aa-properties-for-the-childrengrid-items). |
+| Layout              | None                                                                           |                                                                                                                                                                     |
+| Gap                 | None                                                                           |                                                                                                                                                                     |
+
+#### `useSx` hook
+
+A `useSx` hook, inspired from [Dripsy](https://www.dripsy.xyz/apis/useSx#recommendations) can be helpful for targeting some minor usecases where users may want to style custom components or components which aren't directly accessible to be styled (for eg. `contentContainerStyle` on `FlatList`). I don't suspect a lot of usecases of this prop currently so this can be taken up later as en enhancement. Sample usage:
+
+```tsx
+import { useSx } from '@razorpay/blade/utils';
+
+// ...
+
+const sx = useSx();
+
+return (
+  <>
+    <CustomHeader style={sx({ marginBottom: 'spacing.2' })} />
+    <FlatList
+      contentContainerStyle={sx({ marginX: 'spacing.1' })}
+      data={data}
+      renderItem={renderItem}
+    />
+  </>
+);
+```
+
+`useSx` should return a function (named `sx` above) that can accept the same keys supported by `sx` prop.
+
+#### Not doing, yet
+
+The following shouldn't be considered right now:
+
+- `className`, `style` props support _(see [drawbacks and constraints](#drawbacksconstraints))_
+- Grid systems
+- Bleed layout components
+- Stack components (same can be achieved with `Flex`, `Grid`)
+- Separate utility component for hiding (same can be achieved with responsive properties on `display` set to `none`, `flex`, `block`, etc.)
+- Page level layout components (nav, header, sidebar, footer, etc.)
+
+### Roadmap
+
+Since layout components have a huge scope we should target to launch these progressively. This also gives us room to iterate based on feedback from consumers.
+
+Here's a proposed timeline:
+
+1. `Box`, `Flex`, `Grid` component
```

Comment:
We could skip `Box` & deliver `Spacer` instead

Reply:
Updated roadmap in https://github.com/razorpay/blade/pull/950/commits/f4e3b6cebb79e8d1a3b72ed54eb53ed67c55dec4